pandas/core/internals/blocks.py

```diff
@@ -412,7 +414,7 @@ def split_and_operate(self, func, *args, **kwargs) -> list[Block]:
     # Up/Down-casting
 
     @final
-    def coerce_to_target_dtype(self, other) -> Block:
+    def coerce_to_target_dtype(self, other, warn_on_upcast: bool = False) -> Block:
```

Comment:
this could be changed to `raise_on_upcast` when the deprecation is enforced

Reply:
this would also cover 'shift' (xref #53802).  I don't recall that being a part of PDEP6.  Am I remembering wrong?

Reply:
oh i see there's a new kwd, never mind